addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -1989,20 +2002,30 @@ export class OdooEditor extends EventTarget {
         const isRemovableInvisible = (node, noBlocks = true) =>
             !isVisible(node, noBlocks) && !isUnremovable(node);
         const endIsStart = end === start;
-        while (end && isRemovableInvisible(end, false) && !end.contains(range.endContainer)) {
+        while (end && isRemovableInvisible(end, false) && !end.contains(range.endContainer) && !keepEndLi) {
             const parent = end.parentNode;
             end.remove();
             end = parent;
         }
+        if (keepEndLi) {
+            fillEmpty(closestElement(end, 'li'));
+        }
         // Same with the start container
         while (
             start &&
-            isRemovableInvisible(start) &&
+            (isRemovableInvisible(start) || keepEndLi) &&
             !(endIsStart && start.contains(range.startContainer))
         ) {
             const parent = start.parentNode;
             start.remove();
             start = parent;
+            if (
+                keepEndLi &&
+                (!['UL', 'OL', 'LI'].includes(start.nodeName) ||
+                (start.hasChildNodes() && !isZWS(start)))
+            ) {
+                keepEndLi = false;
+            }
```

Comment:
Does it really need to be in the loop? I mean we know it's going to be the first (or maybe last?) node, right? Can't we do that before or after the loop? So that `keepEndLi` could be a `const` rather than a `let`.

Reply:
We need to keep the loop because, if the start is a nested list, we need to remove all elements till the `li.oe-nested` element is not removed.

Reply:
I think the problem has been misunderstood. It's merely a particular case where both the start LI and the end LI are fully selected and they come from different lists. This should be detectable before doing altering the list and saved into a variable. Then let the code run its course as normal and the result will be almost what we want, it's just that the empty Li will be in the first list rather than be in the second one. We can simply look at the variable we stored before to see if we're in the special case, if yes, we just move the empty li in the second list. Something like that, in pseudo code:
```js
const isItTheSpecialCase = start fully selected and end fully selected and they are different list types
... normal code, with the doJoin etc etc
if (isItTheSpecialCase) {
    move current empty LI (the one that has the selection) to the start of the second list rather than the first
}
```
I have a feeling this would be better for isolating the handling of that special case from the remaining of the logic which is not meant to handle it. It's really a special case because it's the only delete operation I know where your cursor ends up in what was the endContainer rather than the startContainer. That is why I think it would have better to handle it separately rather than trying to incorporate it in the rest of the logic which follows a different principle.

Tell me if it works!

Reply:
Instead of checking if the start and end `LI` are fully selected and then moving the start `LI` to the beginning of the end `LI`, how about doing something like this that I have done here?
https://github.com/odoo-dev/odoo/commit/d5e7d7bfe70c2b0c886ba42c58d822ad19996123

Reply:
I like the code at https://github.com/odoo-dev/odoo/commit/d5e7d7bfe70c2b0c886ba42c58d822ad19996123

Reply:
applied the fix from the https://github.com/odoo-dev/odoo/commit/d5e7d7bfe70c2b0c886ba42c58d822ad19996123 commit and also made some changes to handle the following case and add a test case for it: `<ul><li>ab</li><li>[cd</li></ul><ol><li>e]f</li><li>gh</li></ol>`